great_expectations/data_context/data_context/abstract_data_context.py

```diff
@@ -2685,17 +2700,10 @@ def get_available_data_asset_names(  # noqa: PLR0912
         """
         data_asset_names = {}
         fluent_data_asset_names = {}
-        if datasource_names is None:
-            datasource_names = [
-                datasource["name"] for datasource in self.list_datasources()
-            ]
-        elif isinstance(datasource_names, str):
-            datasource_names = [datasource_names]
-        elif not isinstance(datasource_names, list):
-            raise ValueError(
-                "Datasource names must be a datasource name, list of datasource names or None (to list all datasources)"
-            )
+        datasource_names = self._validate_datasource_names(datasource_names)
 
+        # TODO: V1-222 batch_kwargs_generator_names is legacy and should be removed for V1
+        # TODO: conditional FDS vs BDS datasource logic should be removed for V1
```

Comment:
This is only runtime change we made.
We started to do a lot more but most of the extra complexity in this function is caused by conditional `FDS` vs `BDS` checking and or related to the legacy ` batch_kwargs_generator_names` usage.